src/HydraAuction/OnChain/FeeEscrow.hs

```diff
@@ -0,0 +1,57 @@
+{-# OPTIONS_GHC -fno-specialise #-}
+
+module HydraAuction.OnChain.FeeEscrow (mkFeeEscrowValidator) where
+
+-- Prelude imports
+import PlutusTx.Prelude
+
+-- import Prelude (quot)
+
+-- Plutus imports
+import Plutus.V1.Ledger.Address (pubKeyHashAddress, scriptHashAddress)
+import Plutus.V1.Ledger.Value (valueOf)
+import Plutus.V2.Ledger.Api (TxInfo (..), TxOut (..), Value, adaSymbol, adaToken, scriptContextTxInfo, txInInfoResolved)
+import Plutus.V2.Ledger.Contexts (ScriptContext, ownHash)
+
+-- Hydra auction imports
+import HydraAuction.Types (
+  AuctionTerms (..),
+  FeeEscrowDatum,
+  FeeEscrowRedeemer (..),
+ )
+import HydraAuctionUtils.Plutus (
+  byAddress,
+  nothingForged,
+ )
+
+{-# INLINEABLE mkFeeEscrowValidator #-}
+mkFeeEscrowValidator :: AuctionTerms -> FeeEscrowDatum -> FeeEscrowRedeemer -> ScriptContext -> Bool
```

Comment:
This is implemented slightly differently as discussed on the spec, as the suggested approach has a few complications. I'll raise this in the meeting today, and will update the spec as well if the changes I propose make sense